### DISCUSSION TITLE
'has-*' modifier not working on label elements

### DISCUSSION THREAD

**Discussion by JorensM** [tailwindlabs/tailwindcss#12927] [Help]
**What version of Tailwind CSS are you using?**

3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

Svelte

**What version of Node.js are you using?**

20.8.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**Reproduction URL**

https://play.tailwindcss.com/D7TH0Gb3vV

If you try changing the `label` to a `div`, you will see that it displays correctly.

**Describe your issue**

When trying to apply a `has-*` modifier to a label element it doesn't work. At first I thought it had something to do with label being an inline element, but even after changing it to `flex`, the issue still persists.

**Comment by adamwathan** [comment]
Hey it is working but when you make it `flex`, the input inside of it expands to fill the label so you can't see the label's background color because it is covered by the input.

Here you can see both `label` and `div` do exactly the same thing when set to `block`:

https://play.tailwindcss.com/SUUBLuwGCX

Here I've set the input to `opacity-50` so you can see that the label _does_ have a black background as expected, it's just being masked by the input:

https://play.tailwindcss.com/WalLxjutqZ

**Reply by JorensM** [reply]
Hi, thanks for confirming that it's not a bug with Tailwind. Could you please help me solve this issue? In my app when I try to use the `has` modifier it doesn't seem to work. Here is the code:

```
<label class="flex cursor-pointer items-center justify-between has-[:disabled]:invisible">
	<div class="mr-1 text-sm text-gray-900">
		{$_('walkin')}
	</div>
	<input disabled={disabled || true} bind:checked={walkin} type="checkbox" class="peer sr-only" />
	<Toggle />
</label>
```

As you can see I'm trying to make the `label` element invisible if it's containing input is disabled, and you can also see that I've set the `disabled` prop to be `true`. Yet my label is not invisible.

Thanks.

**Reply by JorensM** [reply]
It seems that this happens if I have `sr-only` added to the input.

https://play.tailwindcss.com/rnaXJVXl0C

I'm not sure if that is the expected behavior, but if yes, is there any workaround to this?

**Reply by JorensM** [reply]
Nevermind, it was the same thing as in your original reply, with the `label`'s size being 0. But for some reason the `has` modifier is not working in my app.
